File: fapacapp/src/app/service/i-service.ts
import { Observable } from "rxjs";
import { RequisicaoPaginada } from "../model/RequisicaoPaginada";
import { RespostaPaginada } from "../model/RespostaPaginada";

export interface IService<T> {
    apiUrl: string;
    get(termoBusca?: string, paginacao?: RequisicaoPaginada): Observable<RespostaPaginada<T>>;
    getById(id: number): Observable<T>;
    save(objeto: T): Observable<T>;
    delete(id: number): Observable<void>;
}

File: fapacapi/src/main/java/fapacapi/service/IService.java
package fapacapi.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface IService<T> {

    public Page<T> get(String termoBusca, Pageable page);
    public T get(Long id);
    public T save (T objeto);
    public void delete(Long id);

}
